test/TestCaseReader.cpp

```diff
@@ -137,6 +144,41 @@ pair<SourceMap, size_t> TestCaseReader::parseSourcesAndSettingsWithLineNumber(is
 				if (sources.count(currentSourceName))
 					BOOST_THROW_EXCEPTION(runtime_error("Multiple definitions of test source \"" + currentSourceName + "\"."));
 			}
+			else if (boost::algorithm::starts_with(line, externalSourceDelimiterStart) && boost::algorithm::ends_with(line, sourceDelimiterEnd))
+			{
+				string externalSourceString = boost::trim_copy(line.substr(
+					externalSourceDelimiterStart.size(),
+					line.size() - sourceDelimiterEnd.size() - externalSourceDelimiterStart.size()
+				));
+
+				string externalSourceName;
+				size_t remappingPos = externalSourceString.find('=');
+				// Does the external source define a remapping?
+				if (remappingPos != string::npos)
+				{
+					externalSourceName = boost::trim_copy(externalSourceString.substr(0, remappingPos));
+					externalSourceString = boost::trim_copy(externalSourceString.substr(remappingPos + 1));
+				}
+				else
+					externalSourceName = externalSourceString;
+
+				solAssert(!externalSourceName.empty(), "");
+				fs::path externalSourceTarget(fs::path{externalSourceString});
```

Comment:
You don't need to construct a `path` object to be able to pass it to the `path` constructor :)
```suggestion
				fs::path externalSourceTarget(externalSourceString);
```